nucypher/characters/lawful.py

```diff
@@ -1364,6 +1367,55 @@ def as_external_validator(self) -> Validator:
         )
         return validator
 
+    def _authorize_lingo(self, decryption_request: ThresholdDecryptionRequest) -> None:
+        # requester-supplied condition eval context
+        context = None
+        if decryption_request.context:
+            # nucypher_core.Context -> str -> dict
+            context = json.loads(str(decryption_request.context)) or dict()
+
+        # obtain condition from request
+        condition_lingo = json.loads(
+            str(decryption_request.acp.conditions)
+        )  # nucypher_core.Conditions -> str -> Lingo
+        if not condition_lingo:
+            # this should never happen for CBD - defeats the purpose
+            raise self.UnauthorizedRequest(
+                "No conditions present for ciphertext - invalid for CBD functionality",
+            )
+
+        # evaluate the conditions for this ciphertext
+        error = evaluate_condition_lingo(
+            condition_lingo=condition_lingo,
+            context=context,
+            providers=self.condition_providers,
+        )
+        if error:
+            raise self.UnauthorizedRequest(
+                f"Condition evaluation failed: {error}",
+            )
+        return condition_lingo
+
+    def _authorize_decryption_request(
+        self, decryption_request: ThresholdDecryptionRequest
+    ) -> None:
+        super()._authorize_decryption_request(decryption_request)
+        self._authorize_lingo(decryption_request)
```

Comment:
What about calling `_authorize_lingo(...)` before calling `self._derive_decryption_share_request(...)` below, then you don't need to override `_authorize_decryption_request`?

Reply:
I attempted to integrate this logic into the existing layers by keeping each block as local to it's dependencies as possible.   This essentially resulted in blockchain interactions on `Operator` and high-level `ThresholdDecryptionRequest|Response` handling on `Ursula`.  Also, by embedding the authorization call inside the derivation it's more misuse resistant -- `derive_decryption_share_for_request(decryption_request)` handles everything (I suppose we can consider this a public API since it's directly exposed by the webserver).

Reply:
Ah Interesting. Given that philosophy, since evaluating condition lingo requires blockchain calls, would it make sense to move `authorize_lingo` functionality into actor.py as well?

Reply:
to be honest I'm not sure it's a valid approach it turned out imperfect and had abstraction leaks that I cannot identify yet.  The main reason for moving lingo here is that is that condition providers is local to Ursula. 🤔

Reply:
Ah yes, you're correct! I'll attempt a slight refactor

Reply:
✔️